provider/googleProvider.js

```diff
@@ -92,6 +92,9 @@ class GoogleProvider {
           },
           additionalProperties: false,
         },
+        cloudFunctionSecretEnvironmentVariables: {
+          type: 'object',
```

Comment:
I'm not across the schema handler here. Is it possible to specify `array` and more validations? Google expects an array of strings like `['SOME_VAR=secret-id:latest']`.

The `serverless.ts` config itself I've left as an object as that allowed for easier overwrite support for each function.